File: java/src/leetcode/P0509.java
package leetcode;

public class P0509 {
    public static void main(String[] args) {
        System.out.println(fib(4));
    }


    public static int fib(int N) {
        if (N == 0) {
            return 0;
        }
        int[] memo = new int[31];
        memo[0] = 1;
        memo[1] = 1;
        for (int i = 2; i < N; i++) {
            memo[i] = memo[i-1] + memo[i-2];
        }
        return memo[N-1];
    }
}

File: ruby/leetcode/problems/P0509.rb
# 509. Fibonacci Number
#
# @param {Integer} n
# @return {Integer}
def fib(n)
  memo = [31]
  memo[0] = 0
  memo[1] = 1
  n.times do |idx|
    if memo[idx].nil?
      memo[idx] = memo[idx - 1] + memo[idx - 2]
    end
  end
  memo[n]
end


5.times do |n|
  puts "hi #{n}"
end
